### ISSUE TITLE
Wine d3d9 visual test fails test_blend() on R600

### ISSUE THREAD

**Issue by iiv3** [iXit/Mesa-3D#125]
When running wine-1.7.49 d3d9 visual test, one of the results is:
`visual.c:8350: Test failed: DSTALPHA on texture returned color 007f0080, expected 0x000000ff`

The test works OK if I use `D3D_ALWAYS_SOFTWARE` or the native wine OpenGL wrapper. From @axeldavy log I see that he also pass that test on his Radeon HD7700. It fails when using Nine on my Radeon HD5670. Unlike #119 `R600_DEBUG=nosb` doesn't fix the problem.

I strongly suspect that it is either R600 compiler or hardware bug.

To help with tracking the problem, I've disabled all other tests in `wine/dlls/d3d9/tests/visual.c` leaving only `test_blend()`, then run it with the command:
`NINE_DEBUG=all NINE_TGSI_DUMP=1 NINE_FF_DUMP=1 R600_DEBUG=vs,ps ./wine dlls/d3d9/tests/d3d9_test.exe.so visual`
The output log can be found here:
[Wine Blend Test](https://gist.github.com/iiv3/1ef6abdb9522aa46018f#file-wine_nine_visual_blend_test)

**Comment by siro20** [comment]
Known Issue on R600.
The destination doesn't have an alpha channel.
In this case the driver should use "D3DBLEND_ONE" instead.
Already fixed on other drivers:
http://lists.freedesktop.org/archives/mesa-commit/2010-May/020647.html
http://lists.freedesktop.org/archives/mesa-dev/2013-January/032575.html
http://lists.freedesktop.org/archives/mesa-dev/2013-February/034786.html

**Comment by iiv3** [comment]
Yes, this bugs seem to have been discussed on [iXit IRC](http://people.freedesktop.org/~cbrill/dri-log/?channel=d3d9&highlight_names=&date=2015-02-12) before.
At the time it's been decided that fixing it would cause too much speed penalty, while there is no real world example of it causing problems.

However #126 seems like a good candidate for such real world example. It fails at blending and same sample works on radeon si. Even #127 is suspected to be linked to it, because it also fails only on R600.

**Comment by siro20** [comment]
https://github.com/siro20/Mesa-3D/commit/b01d1ceb7447ab3cafe8d097864e74203b3843bc
hacks around this issue
but should be fixed in R600
It seems not to fix any other issue.

**Comment by okias** [comment]
@siro20 did you tried ask @mareko to fix it in r600 driver?

**Comment by stalkerg** [comment]
We have answer from @mareko ?
